bindings/go/descriptor/normalisation/engine/jcs/doc.go

```diff
@@ -0,0 +1,91 @@
+// Package jcs implements JSON Canonicalization Scheme (JCS) as defined in RFC 8785.
+// JCS is a deterministic transformation of JSON documents that ensures that logically
+// equivalent documents have the same byte representation.
+//
+// The package provides a comprehensive set of tools for normalizing JSON data:
+// - JSON canonicalization according to RFC 8785
+// - Flexible exclusion rules for filtering JSON structures
+// - Support for custom value mapping and transformation
+// - Handling of special cases like empty values and null fields
+//
+// Key Features:
+// - Implements the complete JCS specification from RFC 8785
+// - Supports both map and array structures
+// - Provides configurable exclusion rules
+// - Handles nested JSON structures
+// - Maintains consistent ordering of map keys
+// - Preserves JSON data types
+//
+// Exclusion Rules:
+// The package provides several types of exclusion rules for fine-grained control over normalization:
+//
+//  1. MapExcludes: Excludes specific fields from maps
+//     excludes := MapExcludes{
+//     "field1": nil,  // Exclude field1
+//     "field2": NoExcludes{},  // Include field2
+//     }
+//
+//  2. MapIncludes: Only includes specified fields from maps
+//     includes := MapIncludes{
+//     "field1": NoExcludes{},  // Include field1
+//     "field2": NoExcludes{},  // Include field2
+//     }
+//
+//  3. ArrayExcludes: Applies rules to all elements in an array
+//     excludes := ArrayExcludes{
+//     Continue: NoExcludes{},  // Rules for array elements
+//     }
+//
+//  4. DynamicArrayExcludes: Applies dynamic rules to array elements
+//     excludes := DynamicArrayExcludes{
+//     ValueChecker: func(v interface{}) bool { return v == nil },
+//     ValueMapper:  func(v interface{}) interface{} { return v },
+//     Continue:     NoExcludes{},
+//     }
+//
+//  5. ExcludeEmpty: Removes empty values (nil, empty maps, empty arrays)
+//     excludes := ExcludeEmpty{ExcludeRules: NoExcludes{}}
+//
+//  6. MapValue: Transforms values before applying exclusion rules
+//     mapper := MapValue{
+//     Mapping: func(v interface{}) interface{} { return v },
+//     Continue: NoExcludes{},
+//     }
+//
+// Usage:
+//
+//	The package can be used to normalize JSON data in various ways:
+//
+//	import (
+//	    "ocm.software/open-component-model/bindings/go/descriptor/normalisation/engine/jcs"
+//	)
+//
+//	// Basic normalization
+//	normalized, err := jcs.Normalise(input, nil)
+//
+//	// With exclusion rules
+//	excludes := jcs.MapExcludes{
+//	    "field1": nil,  // Exclude field1
+//	    "field2": jcs.NoExcludes{},  // Include field2
```

Comment:
This seems weird. If I don't want that field excluded, I should not add it to the `jcs.MaxExcludes{}`. What am I missing?

Reply:
You are correct but there are two ways on this. NoExcludes is an exclusion rule like any other. So semantically "field2": jcs.NoExcludes{} is even equivalent to not setting the field. NoExcludes is mostly used within the engine itself to return if no fields should be excluded as part of `Field(name string, value interface{}) (string, interface{}, ExcludeRules)` which expects to return exclusion rules. I can remove this from the example.

Reply:
ah, I see - thanks for clarifying. Maybe this can lead to confusion in the documentation without any hint (like the comment you just gave). Thus, I would either clarify it or omit it.